modules/Attendance/attendance.php

```diff
@@ -16,488 +16,447 @@
+                }
+
+                // define DataTable
+                $takeAttendanceURL = '/modules/Attendance/attendance_take_byRollGroup.php';
+                $attendanceByCourseClassTable = $getDailyAttendanceTable(
+                    $guid,
+                    $connection2,
+                    'gibbonCourseClassID',
+                    $takeAttendanceURL
+                );
+                $attendanceByCourseClassTable->setTitle(__('My Classes'));
+                $attendanceByCourseClassTable->withData(new DataSet($attendanceByCourseClass));
+            }
+        }
+    }
+}
 
-		                                // Wrap to a new line every 10 dates
-		                                if (  ($historyCount+1) % 10 == 0 ) {
-		                                    echo '</tr><tr>';
-		                                }
-
-		                                $historyCount++;
-		                            }
-		                        }
-
-		                        echo '</tr>';
-		                        echo '</table>';
-
-								print "</td>" ;
+?>
 
-								print "<td style='text-align: center'>" ;
-								// Attendance not taken, timetabled
-								if (isset($ttHistory[$row['gibbonCourseClassID']][$currentDate]) == true) {
-									if ($resultLog->rowCount()<1) {
-										print '<img src="./themes/' . $_SESSION[$guid]["gibbonThemeName"] . '/img/iconCross.png"/>' ;
-									} else {
-										print '<img src="./themes/' . $_SESSION[$guid]["gibbonThemeName"] . '/img/iconTick.png"/>' ;
-									}
-								} else if (isset($logHistory[$row['gibbonCourseClassID']][$currentDate])) {
-                                    echo '<span title="'.__('This class is not timetabled to run on the specified date. Attendance may still be taken for this group however it currently falls outside the regular schedule for this class.').'">';
-                                        echo __('N/A');
-                                    echo '</span>';
-                                }
-								print "</td>" ;
-
-								print "<td style='text-align: center'>" ;
-									print ($resultLog->rowCount()<1)? "" : ($log["total"] - $log["absent"]);
-								print "</td>" ;
-
-								print "<td style='text-align: center'>" ;
-									print $log["absent"];
-								print "</td>" ;
-
-
-								if (isActionAccessible($guid, $connection2, "/modules/Attendance/attendance_take_byCourseClass.php")) {
-
-									print "<td style='text-align: center'>" ;
-										print "<a href='index.php?q=/modules/Attendance/attendance_take_byCourseClass.php&gibbonCourseClassID=" . $row["gibbonCourseClassID"] . "&currentDate=" . dateConvertBack($guid, $currentDate) . "'><img title='" . __('Take Attendance') . "' src='./themes/" . $_SESSION[$guid]["gibbonThemeName"] . "/img/attendance.png'/></a>" ;
-									print "</td>" ;
-
-								}
-
-
-							print "</tr>" ;
-						}
-					print "</table>" ;
-				}
-			}
-		}
-	}
+<?php if (isset($form)) {
```

Comment:
Getting there! You've removed some of the `<?php` `?>` escaping, but I wonder why not all of them? There's no HTML or templating here, so we likely don't need to escape from php execution at all.